src/App.tsx

```diff
@@ -14,11 +15,75 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGood: string[],
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGood: ['Jam'],
+  };
+
+  isSelected = (good: string, allGoods: string[]) => {
+    this.setState({ selectedGood: [...allGoods, good] });
+  };
```

Comment:
or it's better to use functional setState with prevState
https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

```suggestion
  selectGood = (good: string) => {
    this.setState({ selectedGoods: [this.state.selectedGoods, good] });
  };
```